### PR TITLE
Core/Scripting:Fix gossips not load from db

### PR DESCRIPTION
**Pull request by leelf00**
**Changes proposed:**

Fix gossip option not load from db.
- [x] Blackrock Depths
- [x] Karazhan
- [x] Battle for Mount Hyjal
- [x] Zulfarrak
- [x] Trail of the Champion 
- [x] Halls of Stone  
- [x] Quest The Hunt is On (ID 11794)
- [x] Quest Last Rites (ID 12109)
- [x] Magisters Terrace
- [x] Escape from DurnholdeKeep
- [x] Serpent Shrine Cavern

  
updates https://github.com/TrinityCore/TrinityCore/issues/22696

**Tests performed:**

build and tested with zhTW client in game.

### CODE REVIEW

**Comment by MaxtorCoder** [comment]
Please rename the sql file to `9999_99_99_world.sql`, the person who will merge this PR will rename it correctly.

**Comment by leelf00** [comment]
updated filename

File: sql/updates/world/3.3.5/9999_99_99_world.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+-- Battle Of Mount Hyjal missing optionid
+DELETE FROM gossip_menu_option where MenuID=7552 and OptionID in (0,1,2);
+DELETE FROM gossip_menu_option where MenuID=7581 and OptionID in (0,1,2);
+DELETE FROM gossip_menu_option where MenuID=7706 and OptionID=0;
```
[the remark this answers is not in the document]
Comment (leelf00) [reply sql/updates/world/3.3.5/9999_99_99_world.sql]: updated

File: sql/updates/world/3.3.5/9999_99_99_world.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
+-- Battle Of Mount Hyjal missing optionid
+DELETE FROM gossip_menu_option WHERE MenuID=7552 AND OptionID IN (0,1,2);
+DELETE FROM gossip_menu_option WHERE MenuID=7581 AND OptionID IN (0,1,2);
+DELETE FROM gossip_menu_option WHERE MenuID=7706 AND OptionID=0;
+INSERT INTO `gossip_menu_option`(`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES (7552, 0, 0, 'My companions and I are with you, Lady Proudmoore.', 14698, 0, 0, 7556, 0, 0, 0, NULL, 0, 26365),(7552, 1, 0, 'We are ready for whatever Archimonde might send our way, Lady Proudmoore.', 15393, 0, 0, 0, 0, 0, 0, NULL, 0, 0),(7552, 2, 0, 'Until we meet again, Lady Proudmoore.', 15416, 0, 0, 0, 0, 0, 0, NULL, 0, 0),(7581, 0, 0, 'We have nothing to fear.', 15445, 0, 0, 0, 0, 0, 0, NULL, 0, 0),(7581, 1, 0, 'I am with you, Thrall.', 14875, 0, 0, 0, 0, 0, 0, NULL, 0, 0),(7581, 2, 0, 'Until we meet again, Thrall.', 15449, 0, 0, 0, 0, 0, 0, NULL, 0, 0),(7706, 0, 0, 'I would be grateful for any aid you can provide, Priestess.', 20242, 0, 0, 0, 0, 0, 0, NULL, 0, 0);
```
Comment (Aokromes) [inline sql/updates/world/3.3.5/9999_99_99_world.sql:5]: add new row after VALUES

File: sql/updates/world/3.3.5/9999_99_99_world.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+DELETE FROM gossip_menu_option WHERE MenuID=7422 AND OptionID=0;
+INSERT INTO `gossip_menu_option`(`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES (7421, 0, 0, 'I\'m not an actor.', 14052, 0, 0, 0, 0, 0, 0, NULL, 0, 0);
+INSERT INTO `gossip_menu_option`(`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES (7422, 0, 0, 'Ok, I\'ll give it a try, then.', 14054, 0, 0, 0, 0, 0, 0, NULL, 0, 0);
+
+-- HallsOfStone missing optionid
+DELETE FROM gossip_menu_option WHERE MenuID=9670 AND OptionID=0;
+INSERT INTO `gossip_menu_option`(`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES (9670, 0, 0, 'Let\'s move Brann, enough of the history lesson!', 27616, 0, 0, 0, 0, 0, 0, NULL, 0, 0);
```
Comment (Aokromes) [inline sql/updates/world/3.3.5/9999_99_99_world.sql:15]: missing newline.

File: sql/updates/world/3.3.5/9999_99_99_world.sql [hunk 4]
```diff
@@ -0,0 +1,24 @@
+-- Battle Of Mount Hyjal missing optionid
+DELETE FROM gossip_menu_option WHERE MenuID=7552 AND OptionID IN (0,1,2);
+DELETE FROM gossip_menu_option WHERE MenuID=7581 AND OptionID IN (0,1,2);
+DELETE FROM gossip_menu_option WHERE MenuID=7706 AND OptionID=0;
+INSERT INTO `gossip_menu_option`(`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES
```
Comment (Aokromes) [inline sql/updates/world/3.3.5/9999_99_99_world.sql:5]: remove ghost whitespace after VALUES :)

File: sql/updates/world/3.3.5/9999_99_99_world.sql [hunk 5]
```diff
@@ -14,11 +14,11 @@ INSERT INTO `gossip_menu_option`(`MenuID`, `OptionID`, `OptionIcon`, `OptionText
...
 (7421, 0, 0, 'I\'m not an actor.', 14052, 0, 0, 0, 0, 0, 0, NULL, 0, 0),
 (7422, 0, 0, 'Ok, I\'ll give it a try, then.', 14054, 0, 0, 0, 0, 0, 0, NULL, 0, 0);
 
 -- HallsOfStone missing optionid
 DELETE FROM gossip_menu_option WHERE MenuID=9670 AND OptionID=0;
-INSERT INTO `gossip_menu_option`(`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES 
+INSERT INTO `gossip_menu_option`(`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES
 (9670, 0, 0, 'Let\'s move Brann, enough of the history lesson!', 27616, 0, 0, 0, 0, 0, 0, NULL, 0, 0);
```
Comment (Aokromes) [inline sql/updates/world/3.3.5/9999_99_99_world.sql:24]: still missing newline :)

Comment (leelf00) [reply sql/updates/world/3.3.5/9999_99_99_world.sql:24]: add newline before insert?

Comment (Aokromes) [reply sql/updates/world/3.3.5/9999_99_99_world.sql:24]: no at end of file.

**Comment by Aokromes** [comment]
it looks good to me.

**Review by Aokromes — APPROVED** [review]

**Comment by leelf00** [comment]
thank for adjuest，but Magisters Terrace       code come back to #define.

**Comment by offl** [comment]
That script used non-existing gossips

**Comment by Aokromes** [comment]
updated https://github.com/TrinityCore/TrinityCore/issues/22696

**Comment by leelf00** [comment]
Magisters Terrace Gossips seems belong to npc Kalecgos (https://tbc.wowhead.com/npc=24848/kalecgos#comments).I check menuid exists in gossip_menu_option,any other thing missing?

**Pull request merged**